README.md

```diff
@@ -77,43 +77,50 @@ To handle the response:
-
-### Encrypting communications
-Once you have created your request in form of `Message<Request>`, you can encrypt it as follows (this applies to `PaymentRequest`):
-```swift
-let encryptionKey: EncryptionKey = // your key derived earlier
-let request: Message<PaymentRequest> = // the payment request previously created
-let encryptedMessage: Data = try request.encrypt(using: encryptionKey)
-```
-The `encryptedMessage` is now ready to be sent to the terminal.
-
-When you have received a response from the terminal, do the following:
-```swift
-let key: EncryptionKey = // key you derived earlier
-let response: Data = // the response you receive from the terminal
-let encryptedMessage: EncryptedMessage = try Coder.decode(EncryptedMessage.self, from: response)
-let decryptedMessage: Message<PaymentResponse> = try decrypt(PaymentRequest.self, using: key)
-```
+To help you with this, TerminalAPIKit provides helper functions to:
+- Derive the encryption key from the [shared key set up in your Adyen Customer Area](https://docs.adyen.com/point-of-sale/choose-your-architecture/local#set-up-shared-key). 
+- Encrypt and decrypt the messages passed between your terminal and your iOS or macOS POS application.
+
+### Derive the encryption key
+To derive the key used for encrypting and decrypting local communications between the terminal and your POS app:
+
+1. Get the `identifier`, `passphrase`, and `version` of the shared key: <br>
+   In the Adyen Customer Area, under **Point of sale**, go to the terminal settings for your merchant account or store. 
+   Select **Integrations** and under **Terminal API** go to **Encryption key**. 
+   To see the key identifier, passphrase, and version values, select **Decrypted**.
+
+2. Derive the key, making sure to pass the values in string form exactly as they appear in the Customer Area.
+   ```swift
+   let encryptionKey = try EncryptionKey(
+       identifier: "KEY_IDENTIFIER",
+       passphrase: "KEY_PASSPHRASE",
+       version: KEY_VERSION
+   )
+   ``` 
+Once the key is derived, you are ready to encrypt your local communications.
+
+### Encrypt and decrypt communications
+1. Create your request as described above. For example, create a `Message<PaymentRequest>`.
+2. Encrypt your request.
+   ```swift
+   let encryptionKey: EncryptionKey = // the key you derived earlier
+   let request: Message<PaymentRequest> = // the payment request you created
+   let encryptedMessage: Data = try request.encrypt(using: encryptionKey)
+   ``` 
+3. Send the `encryptedMessage` to the terminal.
+4. When you receive the response from the terminal, decrypt the response.
+   ```swift
+   let key: EncryptionKey = // the key you derived earlier
+   let response: Data = // the response you receive from the terminal
+   let encryptedMessage: EncryptedMessage = try Coder.decode(EncryptedMessage.self, from: response)
+   let decryptedMessage: Message<PaymentResponse> = try decrypt(PaymentRequest.self, using: key)
```

Comment:
this line should be
`let decryptedMessage: Message<PaymentResponse> = try decrypt(PaymentResponse.self, using: key)`